gym_ignition/base/robot/robot_links.py

```diff
@@ -26,19 +26,55 @@ def link_names(self) -> List[str]:
         """
 
     @abstractmethod
-    def link_pose(self, link_name: str) -> np.ndarray:
+    def link_pose(self, link_name: str) -> Tuple[np.ndarray, np.ndarray]:
         """
         Return the pose of the specified link.
 
         The pose is composed of a position and a quaternion associated to the link frame,
-        both expressed in the the robot base frame.
+        both expressed in the the world frame.
 
         Args:
             link_name: The name of the link
 
         Returns:
-            A Tuple containing the position and orientation of the base:
+            A Tuple containing the position and orientation of the link:
 
             - position: 3D array in the [x, y, z] form.
             - orientation: a 4D array containing a quaternion in the [w, x, y, z] form.
         """
+
+    @abstractmethod
+    def link_velocity(self, link_name: str) -> Tuple[np.ndarray, np.ndarray]:
+        """
+        Return the velocity of the specified link.
+
+        The velocity is composed of linear and angular velocities of the link frame,
+        both expressed in the world frame.
+
+        Args:
+            link_name: The name of the link
+
+        Returns:
+            A Tuple containing the linear and angular velocity of the link:
+
+            - linear velocity: a 3D array in the [vx, vy, vz] form.
+            - angular velocity: a 3D array in the [wx, wy, wz] form.
+        """
+
+    @abstractmethod
+    def link_acceleration(self, link_name: str) -> Tuple[np.ndarray, np.ndarray]:
+        """
+        Return the acceleration of the specified link.
+
+        The acceleration is composed of linear and angular accelerations of the link
+        frame, both expressed in the world frame.
```

Comment:
In which frame the derivation is done?

Reply:
What do you mean more specifically? The method returns the link acceleration in the world frame.

Reply:
Sorry for being vague. The point is that "expressed in the world frame" is quite ambiguous. Let's say that $A$ is an absolute/world frame and $B$ is a body frame, and ${}^A o_B \in \mathbb{R}^3$, ${}^A R_B$ and  ${}^A \omega_{A,B}$ have the meaning defined  in Section 4.3 of  https://pure.tue.nl/ws/portalfiles/portal/139293126/A_Multibody_Dynamics_Notation_Revision_2_.pdf: then "linear and angular acceleration expressed in the world frame" could mean: 

$$
\begin{bmatrix}
{}^A \ddot{o}_B \\
{}^A \dot{\omega}_{A,B}
\end{bmatrix}
$$

for somebody used to the conventions used in multi-task control (we sometime call this "mixed" acceleration) 

or 

$$
\begin{bmatrix}
\frac{d}{dt} ( {}^A \dot{o}_B -  {}^A \dot{\omega}_{A,B} \times {}^A o_B ) \\
{}^A \dot{\omega}_{A,B}
\end{bmatrix}
$$

for somebody used to the Featherston book or Featherstone's  notation of "Spatial vector" (we sometime call this "right-trivialized" or "world" acceleration)

Reply:
A bit longer explanation on the different kind of accelerations is available in the Chapter 2 of my PhD Thesis (https://traversaro.github.io/phd-thesis/traversaro-phd-thesis.pdf).

Reply:
Thank you @traversaro for the explanation and links. I tried to adapt https://github.com/robotology/gym-ignition/pull/117#discussion_r371005743 also for the acceleration. This is another piece of information that should go in the new documentation #37